docs/source-app/workflows/add_web_ui/panel/basic.rst

```diff
@@ -0,0 +1,329 @@
+
+----
+
+1. Define a Panel app
+^^^^^^^^^^^^^^^^^^^^^
+
+First, find the Panel app you want to integrate. In this example, that app looks like:
+
+.. code:: python
+
+    import panel as pn
+    import plotly.express as px
+
+    pn.extension("plotly", sizing_mode="stretch_width")
+
+    ACCENT = "#792EE5"
+
+
+    def get_panel_theme():
+        """Returns 'default' or 'dark'"""
+        return pn.state.session_args.get("theme", [b"default"])[0].decode()
+
+
+    def get_plotly_template():
+        if get_panel_theme() == "dark":
+            return "plotly_dark"
+        return "plotly_white"
+
+        def get_plot(length=5):
+            xseries = [index for index in range(length + 1)]
+            yseries = [x ** 2 for x in xseries]
+            fig = px.line(
+                x=xseries,
+                y=yseries,
+                template=get_plotly_template(),
+                color_discrete_sequence=[ACCENT],
+                range_x=(0, 10),
+                markers=True,
+            )
+            fig.layout.autosize = True
+            return fig
+
+        def get_view():
+            length = pn.widgets.IntSlider(value=5, start=1, end=10, name="Length")
+            plot = pn.bind(get_plot, length=length)
+            component = pn.Column(length, plot)
+            template = pn.template.FastListTemplate(title="⚡ Hello Panel + Lightning ⚡", main=[component], accent=ACCENT)
+            return template
+
+        get_view().servable()
+
+You can serve the app by running
+
+.. code:: bash
+
+    panel serve 'name_of_script.py' --autoreload --show
+
+This Panel app plots a simple line curve depending on the value of a slider.
+Its all wrapped in a nice and easy to use template.
+
+`Visit the Panel documentation for the full API <https://panel.holoviz.org>`_.
+
+----
+
+2. Add the Panel app to a lightning component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add the Panel app to the run method of a ``LightningWork`` component and run the server on that
+component's **host** and **port**.
+
+.. code:: python
+    :emphasize-lines: 7, 8, 10
+
+    class LitPanel(lapp.LightningWork):
+        def __init__(self, **kwargs):
+            super().__init__(**kwargs)
+
+            self.requests = 0
+
+        def _fast_initial_view(self):
+            self.requests += 1
+            if self.requests == 1:
+                return pn.pane.HTML("<h1>Please refresh the browser to see the app.</h1>")
+            else:
+                return get_view()
+
+        def run(self):
+            pn.serve(
+                {"/": self._fast_initial_view},
+                port=self.port,
+                address=self.host,
+                websocket_origin="*",
+                show=False,
+            )
+
+Please note we need to wrap the `view` function by the `_fast_initial_view` such that the first,
+initial request done by the lightning server is fast. Otherwise it will continue requesting a
+response and never load the page.
```

Comment:
I would mention this behavior will likely change in the future and doing this won't be necessary.